### PR TITLE
[stoney] More fixes to make the "disable 15-minute chef-client run" not crash everything

### PR DESCRIPTION
**Pull request by vuntz**

### CODE REVIEW

**Comment by vuntz** [comment]
cc @bkutil, tboerger: I'm not quite sure about the second commit, but `@logger` was definitely creating an issue when used from a forked process. Can you confirm my fix is correct?

**Comment by dirkmueller** [comment]
LGTM

File: crowbar_framework/app/models/service_object.rb [hunk 1]
```diff
@@ -1492,14 +1492,15 @@ def chef_daemon(action, node_list)
...
   def wait_for_chef_clients(node_name, options = {})
-    options = if options.fetch(:logger)
+    use_logger = options.fetch(:logger)
+    options = if use_logger
       {:logger => @logger}
     else
       {}
     end
-    @logger.debug("wait_for_chef_clients: Waiting for already running chef-clients on #{node_name}.")
+    @logger.debug("wait_for_chef_clients: Waiting for already running chef-clients on #{node_name}.") if use_logger
```
Comment (MaximilianMeister) [inline crowbar_framework/app/models/service_object.rb]: I dont understand why this logger shouldn't work in some case

**Comment by MaximilianMeister** [comment]
+1 for the first commit

File: crowbar_framework/app/models/service_object.rb (hunk 1 above)
Comment (vuntz) [inline crowbar_framework/app/models/service_object.rb]: That's what I mentioned in my comment on the PR: it seems `@logger` is not available after the fork. But only on ruby 1.8.

Comment (bkutil) [inline crowbar_framework/app/models/service_object.rb]: What is the reason behind options mangling?

Wouldn't something like this work, considering `wait_for_chef_clients` is called with `options = { :logger => @logger }`?:

```ruby
logger = options.fetch(:logger) { Rails.logger }
logger.debug("wait_for...")
...
options[:logger] ||= logger
unless RemoteNode.chef_ready?(....options)
  logger.error("Waiting...")
```

Comment (vuntz) [inline crowbar_framework/app/models/service_object.rb]: I believe the issue is that `Rails.logger` is kind of useless after the fork, and so should not be used in there. But maybe that works; unfortunately, can't try that right now.

**Comment by vuntz** [comment]
I'd appreciate if people could try to fix the second issue I was fixing another way. I'm happy to take suggestions, since I was not able to find a better fix.

**Comment by dirkmueller** [comment]
@MaximilianMeister , @bkutil , @rsalevsky  does it really matter to fix the 2nd commit which fixes a stoney only issue? this patch is in our packages for ages, its time to get it merged imho. 

it LGTM, but I can't review it.

File: crowbar_framework/app/models/service_object.rb (hunk 1 above)
Comment (MaximilianMeister) [inline crowbar_framework/app/models/service_object.rb]: Usually you use `Rails.logger` outside the models or controllers. but `Rails.logger` must always work in the full scope of the App.

**Comment by MaximilianMeister** [comment]
@vuntz @dirkmueller I dont have any other suggestion. in fact i don't understand why this should be needed or what is the difference (in terms of logic). but if it makes a technical difference and if it fixes the issue then we should consider merging it.

**Comment by rsalevsky** [comment]
I'm fine with merging it, the second commit is a little bit hacky but it works.

**Comment by aspiers** [comment]
:-1: as it stands, sorry.

**Comment by vuntz** [comment]
Obsoleted by https://github.com/crowbar/barclamp-crowbar/pull/1352

**Pull request closed without merging**